app/lib/osm-p2p.js

```diff
@@ -64,9 +72,46 @@ function osmp2p() {
     return osm.query(q, opts, cb);
   }
 
+  function queryStream(q, opts) {
+    return osm.queryStream(q, opts);
+  }
+
+  function queryGeoJSONStream(q, opts) {
+    var osmStream = osm.queryStream(q, opts);
+    var geoJSONStream = getGeoJSON.obj(osm);
+    return pump(osmStream, geoJSONStream);
+  }
+
   function replicate(opts) {
     return osm.log.replicate(opts);
   }
+
+  function sync(transportStream, opts, callback) {
+    if (typeof opts === "function") {
+      callback = opts;
+      opts = null;
+    }
+
+    console.log("sync: start");
+    var osmStream = replicate(opts);
+
+    eos(osmStream, function(err) {
+      if (err) return console.log("osmStream error", err);
```

Comment:
i think this will be temporary until we have some kind of error notification ui we can use to show a message like "Something went wrong with the data sync".